src/main/java/tasklist/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package tasklist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import command.CommandNotFoundException;
+import ouiouibaguette.OuiOuiBaguetteException;
+import storage.Storage;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.ToDo;
+
+/**
+ * Represents a list of tasks and provides methods to manipulate and access these tasks.
+ * The tasks are loaded from and stored in a given storage system.
+ */
+public class TaskList {
+
+    /** The storage system used to persist the tasks. */
+    protected Storage storage;
+
+    /** The list that holds all the tasks. */
+    protected List<Task> tasklist;
```

Comment:
Consider renaming tasklist to tasks for simplicity and clarity, as the List type already implies it’s a collection.